lib/broadway_dashboard.ex

```diff
@@ -19,31 +19,51 @@ defmodule BroadwayDashboard do
 
   @impl true
   def init(opts) do
-    pipelines = Keyword.get(opts, :pipelines, [])
+    pipelines = opts[:pipelines] || :auto_discover
 
     {:ok, %{pipelines: pipelines}, application: :broadway}
   end
 
   @impl true
   def menu_link(%{pipelines: pipelines}, _capabilities) do
-    case pipelines do
-      [] ->
-        {:disabled, @page_title, @disabled_link}
+    if pipelines == [] do
+      {:disabled, @page_title, @disabled_link}
+    else
+      {:ok, @page_title}
+    end
+  end
 
-      [_ | _] ->
-        {:ok, @page_title}
+  defp pipelines_or_auto_discover(pipeline_config, node) do
+    if pipeline_config == :auto_discover do
+      with {:broadway, :ok} <- {:broadway, check_broadway_version(node)},
```

Comment:
I would likely rewrite this to a case, since you have to explicitly handle the results of both clauses.

Reply:
good point! b25aa150c3df0640567f124e4737491f15a62e5a